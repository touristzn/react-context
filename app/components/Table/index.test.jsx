import React from 'react';
import PropTypes from 'prop-types';
import { shallow } from 'enzyme';
import { Table, TableColumn } from './index';

const props = {
  data: [
    {
      "id": 1,
      "title": "去云南旅游",
      "price": 200,
      "date": "2018-08-01",
      "category": {
        "id": 1,
        "name": "旅行",
        "type": "outname",
      },
    },
    {
      "id": 2,
      "title": "去云南旅游",
      "price": 300,
      "date": "2018-08-01",
      "category": {
        "id": 1,
        "name": "旅行",
        "type": "outname",
      },
    },
  ],
  // width: '100%',
};

let wrapper;

describe('test Tabs and Tab Component', () => {
  beforeEach(() => {
    wrapper = shallow(
      <Table {...props}>
        <TableColumn prop="id" width="50" label="序号" />
        <TableColumn prop="title" label="内容" />
        <TableColumn prop="date" width="150" label="日期" />
        <TableColumn>
          <span onClick={() => {}}>修改</span>
          <span onClick={() => {}}>查看</span>
        </TableColumn>
      </Table>
    )
  })

  it('should render the component to match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
  it('should render Two TableColumn component', () => {
    expect(wrapper.find(TableColumn)).toHaveLength(2);
  })
});