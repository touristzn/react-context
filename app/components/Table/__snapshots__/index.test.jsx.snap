// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Tabs and Tab Component should render the component to match the snapshot 1`] = `ShallowWrapper {}`;
